### PR TITLE
#27: FunctionalitiesManager and derived classes

### PR DESCRIPTION
**Pull request by michau-k**
- Functionalities part of the MainController
- Class included : 
  - FunctionalitiesManager
  - AFunctionality

### CODE REVIEW

File: source/common/microkernel_architecture/microkernel_modules/functionalities/FunctionalitiesManager.hpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+public:
+  FunctionalitiesManager();
+  ~FunctionalitiesManager();
+
+  bool init();
+  void shutdown();
+  AFunctionality *loadLibrary(const QString &name);
+
+private:
```
Comment (clairealmozinos) [inline source/common/microkernel_architecture/microkernel_modules/functionalities/FunctionalitiesManager.hpp:28]: There are two section private. Gather them.

File: source/common/microkernel_architecture/microkernel_modules/functionalities/AFunctionality.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+//
+// Start the functionnality, threaded by default
+// and send a signal to FunctionalitiesManager class
+// to keep a trace of what functionalities are running right now
+//
+bool AFunctionality::start(bool threaded)
+{
+  if (_running)
```
Comment (clairealmozinos) [inline source/common/microkernel_architecture/microkernel_modules/functionalities/AFunctionality.cpp:24]: I don't know if it is GitHub but the tabulations seems to be 2 spaces and not 4.

File: source/common/microkernel_architecture/microkernel_modules/functionalities/FunctionalitiesManager.hpp (hunk 1 above)
Comment (Grenadingue) [inline source/common/microkernel_architecture/microkernel_modules/functionalities/FunctionalitiesManager.hpp:28]: That's not in the norm
I do the same myself, please do not consider it wrong

File: source/common/microkernel_architecture/microkernel_modules/functionalities/AFunctionality.cpp (hunk 2 above)
Comment (Grenadingue) [inline source/common/microkernel_architecture/microkernel_modules/functionalities/AFunctionality.cpp:24]: Effectively 2 instead of 4

**Pull request closed without merging**